pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,153 @@
+from pyanaconda.core import util
+from pyanaconda.core.configuration.anaconda import conf
+from pyanaconda.product import productName
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SYSTEMD"]
+
+class SYSTEMD(BootLoader):
+    """Systemd-boot.
+
+    """
+    name = "SYSTEMD"
+    # oddly systemd-boot files are part of the systemd-udev package
+    # and in /usr/lib/systemd/boot/efi/systemd-boot[aa64].efi
+    # and the systemd stubs are in /usr/lib/systemd/linuxaa64.efi.stub
+    _config_file = "loader.conf"
+    _config_dir = "/loader"
+
+    # systemd-boot doesn't require a stage2 as
+    # everything is stored on the ESP
+    stage2_max_end = None
+    stage2_is_valid_stage1 = True
+    stage2_required = False
+
+    #
+    # configuration
+    #
+
+    @property
+    def config_dir(self):
+        """ Full path to configuration directory. """
+        log.info("systemd.py: config_dir")
+        esp = util.execWithCapture("bootctl", [ "--print-esp-path" ],
+                                   root=conf.target.system_root)
+        return esp.strip() + self._config_dir
+
+    @property
+    def config_file(self):
+        """ Full path to configuration file. """
+        log.info("systemd.py: config_file")
+        return "%s/%s" % (self.config_dir, self._config_file)
+
+    # copy console update from grub2.py
+    def write_config_console(self, config):
+        log.info("systemd.py: write_config_console")
+        if not self.console:
+            return
+
+        console_arg = "console=%s" % self.console
+        if self.console_options:
+            console_arg += ",%s" % self.console_options
+        self.boot_args.add(console_arg)
+
+    def write_config(self):
+        log.info("systemd.py: write_config systemd start")
+        self.write_config_console(None)
+
+        # Rewrite the loader.conf
+        # For now we are just updating the timeout to actually
+        # implement the bootloader --timeout option
+        config_path = "%s%s" % (conf.target.system_root, self.config_file)
+        log.info("systemd.py: write_config systemd loader conf : %s ", config_path)
+
+        config = open(config_path, "w")
+        config.write("timeout "+ str(self.timeout) + "\n")
+        config.write("#console-mode keep\n")
+        config.close()
+
+        # update /etc/kernel/cmdline
+        # should look something like "root=UUID=45b931b7-592a-46dc-9c33-d38d5901ec29 ro resume=/dev/sda3"
+        config_path = "%s%s" % (conf.target.system_root, "/etc/kernel/cmdline")
+        log.info("systemd.py: write_config systemd commandline : %s ", config_path)
+        config = open(config_path, "w")
+        args = " ".join(list(self.boot_args._arguments))
+        log.info("systemd.py: systemd used boot args: %s ", args)
+
+        # pick up the UUID of the mounted rootfs,
+        root_uuid = util.execWithCapture("findmnt", [ "-sfn", "-oUUID", "/" ],
```

Comment:
I would expect to get the UUID from the installer, but if it works, I am fine with that.

Reply:
I left this alone, IIRC when I initially wrote this, I tried using a UUID provided elsewhere but was having issues, so this was the trivial/self-contained fix.